compiler/luci/pass/src/FoldSqueezePass.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldSqueezePass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+/**
+ * Fold Squeeze to const if it has const input
+ **/
+bool fold_squeeze(luci::CircleSqueeze *squeeze)
+{
+  // Check squeeze has const input
+  auto const_input = dynamic_cast<luci::CircleConst *>(squeeze->input());
+  if (not const_input)
+    return false;
+
+  // Check all dimensions are known
```

Comment:
Just curiosity, is it possible for the const node to have unknown dimension?

Reply:
> is it possible for the const node to have unknown dimension?

as of now, no